### DISCUSSION TITLE
Challanges order?

### DISCUSSION THREAD

**Discussion by tomer953** [tomalaforge/angular-challenges#154] [Ideas]
Hi,

the current challanges order is kinda confusing (by groups, then by id),
for most of the users the Id is not important if its not related to certain order, making progress, or diffuclty.
IMO it makes more sense (and better UI\UX) that challanges will be orderd by difficulty (green-yellow-red) in each category. wdyt ?

**Comment by tomalaforge** [comment]
Hello, 

Thanks for opening a discussion on that topic. I tried many different approaches but I never had any feedbacks. So thank you for giving your opinion. 

I added categories for better UX and concept separation. But  indeed challenge number is not important to the user. And ordering by difficulty make more sense, I suppose. 

I choose to order the challenge by number to make it simpler to order automatically inside starlight since i know the next challenge number. 

We can try by difficulty and see if we find another option better in the future.

If we order by difficulty, it means we need to have an order strategy. Like easy, starlight order starts at 1, medium at 100 and hard at 200 so it's easy to had a new one. And we can try to automate it by adding the count inside the `challenge-number.json` file.

**Comment by tomer953** [comment]
its ok to start with 1, 100, and 200 but if you want to change also the inner order between easy challanges, it always a good thing to keep a gap, therefor I useuly use:
order: 10, 20, 30 and so on, so inserting a new one in the middle will be easy (calc the avg between the two elements it need to go between)

since new challanges are not that often, all we need is to decide for the gap (10 is fine, 100 is kinda bullet proof)

thoughts?

**Comment by tomalaforge** [comment]
I think I prefer to start the easy challenge at 1, medium at 100 and hard at 200 and just increment that number
If we need to insert a challenge, this will not happen very often. We will change the numbers manually. 

make sense 😅

**Comment by tomalaforge** [comment]
Do you want to do it?

If so, add a counter inside the challenge-number file for each difficulty. And inside the challenge generator `https://github.com/tomalaforge/angular-challenges/blob/main/libs/cli/src/generators/challenge/generator.ts`, we need some logic to automaticaly add a new order to the generated file depending on the chosen difficulty.

If you have a better idea on how to achieve that, help yourself 👍

If you don't want to do it, that's ok, I will create an issue for it. This can be a great first contribution issue.

**Comment by tomer953** [comment]
Yeah I'll try. I see couple of bugs already, so I'll try break into seperate commits and need your review

**Comment by tomalaforge** [comment]
awesome, thanks 💪

**Comment by tomer953** [comment]
You welcome, #155
